index.html

```diff
@@ -727,52 +723,69 @@ <h2>
             </li>
           </ol>
         </li>
-        <li>If |orientation| is `null`, [=unlock the screen orientation=].
-        </li>
-        <li>Otherwise, attempt to [=lock the screen orientation=] to
-        |orientation|. Depending on platform conventions, change how the
-        viewport is drawn to match |orientation|. If the attempt fails due to
-        previously-established user preference, or platform limitation, or any
-        other reason:
+        <li>Run the following sub-steps [=in parallel=]. If |document| stops
+        being [=Document/fully active=] while [=in parallel=], and if the
+        |document|'s {{Document/[[orientationPendingPromise]]}} is not `null`,
+        [=reject and nullify the current lock promise=] of |document| with an
+        {{"AbortError"}}:
```

Comment:
Technically speaking this is not something you can do. You'd have to periodically queue tasks to make this check. But realistically this should all happen so fast that you should just check this once the in parallel algorithm has done its in parallel thing and queues a task to continue?

Reply:
Hmm, interesting... It might be a Gecko-ism, but there it was possible to get notifications of the document changing state (e.g., in payment request,  it was used this to [invalidate various promises](https://searchfox.org/mozilla-central/source/dom/payments/PaymentRequest.cpp#1209)). 

I'll move this though, as you suggested, as you are right that everything happens in quick succession (unlike payment request, where the payment sheet is indefinitely waiting for user input).

Reply:
Thinking out loud...  this is the only case we can realistically capture:

```JS
const iframe = document.querySelector("iframe");
const p = iframe.contentWindows.screen.orientation.lock("landscape");
iframe.remove();
// p aborts
```

So, the check should happen as the very first thing after going "in parallel".

Reply:
It's probably possible to design a system whereby you get notified, but that would then be some kind of thing you register for in the HTML standard and not something you can access when in parallel.

Reply:
Yeah, understood. Do you think the intention is clear enough though? It's not too dissimilar to, for instance, in img decode9() method. It also has similar wording:

> This [img](https://html.spec.whatwg.org/#the-img-element) element's [node document](https://dom.spec.whatwg.org/#concept-node-document) stops being [fully active](https://html.spec.whatwg.org/#fully-active)
>... other cases...
>Reject promise with an ["EncodingError"](https://webidl.spec.whatwg.org/#encodingerror) [DOMException](https://webidl.spec.whatwg.org/#dfn-DOMException).

Reply:
I guess it's good enough, but you should at least queue a task. That's an obvious bug in `decode()` we don't have to mimic.